### ISSUE TITLE
Permissions not taking affect since 1.6.1

### ISSUE THREAD

**Issue by Jonsapps** [realm/realm-mobile-platform#190]
## Goals

I am trying to create a common realm for all users to read and write to

## Expected Results

When connecting to the object server using an admin account I should be able to connect to a common realm e.g "/common/sharedrealm" and grant read and write permission for every user using * as the user id.

## Actual Results

Since updating to 1.6.1 (Object Server) granting permissions from an admin account don't appear to be taking affect and other users are unable to read or write to the shared realm.
The same setup ran against 1.5.0 works as expected.

## Steps to Reproduce

1. Connect to the object server using an admin account and the realm path of "/common/sharedrealm"
2. Once the account has logged in run the below method to grant read and write access to all users
3. Login to the object server webpage and notice the default permissions for the realm are still "no access"
4. Connect to the realm using another account and notice no data is synced

## Code Sample

```swift
func grantReadWriteToAllUsers(user: SyncUser)
    {
        // grant access to every user
        let permission = SyncPermissionValue(realmPath: "/common/\(syncRealmPath)", userID: "*", accessLevel: .write)
        user.applyPermission(permission) { error in
            if let error = error
            {
                print(error.localizedDescription)
                return
            }
        }
    }
```

## Version of Realm and Tooling

- Realm Object Server Version: 1.6.1
- Flavor:
  - [ X] Developer
  - [ ] Professional
  - [ ] Enterprise
- Server OS & Version: macOS 10.12.4
- Client SDK Version: 2.7.0
- Client OS & Version: iOS 10.3

**Comment by ianpward** [comment]
@Jonsapps is the `user` an Admin User with the Admin Token or a user with a login username/password that has been granted admin privileges?

**Comment by Jonsapps** [comment]
This is an account using a username and password who has been granted admin, it's also the same account that you have to make when you first start the object server

**Comment by Jonsapps** [comment]
I have updated to version 1.7.2 and the same issue exists. Although I have found that if I add an entry into the permission model in the _admin realm with a nil user pointing to the realm I want to give default read/write permissions to, the realm dashboard now shows read/write access as the default for my shared realm

**Comment by netspencer** [comment]
yeah. not working for me either :(

**Comment by netspencer** [comment]
i think the issue is with using * as the userId.

`      managementRealm.write(() => {
        managementRealm.create('PermissionChange', {
          id: uuid(),
          createdAt: new Date(),
          updatedAt: new Date(),
          userId: '*',
          realmUrl: realmUrl,
          mayRead: true,
          mayWrite: true
        })
      })
`

**Comment by mrackwitz** [comment]
I could locate this as a regression within the permission management system when using admin users to operate with unowned Realm files (in the global scope, e.g. `/global`). We've a fix and bumped up test coverage around this case. We are working on getting a bug fix release out.

**Comment by mrackwitz** [comment]
We shipped the bug fix with version 1.7.5. Please update to the latest version and let us know whether this resolves the issue.

**Comment by svrzii** [comment]
Hello @mrackwitz, I had the same issue as @Jonsapps and thought that it was fixed. On my Iphone 6s & Ipad mini 2 this works like it should, but on an older device like iphone 5 with iOS 10 & an older Ipad with iOS 9 the user don't get the read & write permissions that are set. @Jonsapps does it work for you on older devices?
